profile-viz/index.html

```diff
@@ -0,0 +1,709 @@
+        }
+
+        .summary-stats {
+            display: grid;
+            grid-template-columns: repeat(auto-fit, minmax(200px, 1fr));
+            gap: 15px;
+            margin-bottom: 20px;
+        }
+
+        .stat-card {
+            padding: 15px;
+            background: linear-gradient(135deg, #667eea 0%, #764ba2 100%);
+            color: white;
+            border-radius: 8px;
+            text-align: center;
+        }
+
+        .stat-value {
+            font-size: 2em;
+            font-weight: bold;
+            margin: 10px 0;
+        }
+
+        .stat-label {
+            font-size: 0.9em;
+            opacity: 0.9;
+        }
+
+        .arrow {
+            display: inline-block;
+            margin-left: 10px;
+            transition: transform 0.3s;
+        }
+
+        .arrow.expanded {
+            transform: rotate(90deg);
+        }
+    </style>
+</head>
+
+<body>
+    <div class="container">
+        <h1>egglog profiler</h1>
+
+        <div class="upload-section">
+            <h2>Upload Profile JSON</h2>
+            <br>
+            <br>
+            <label for="fileInput" class="upload-btn">Choose File</label>
+            <input type="file" id="fileInput" accept=".json">
+            <p id="fileName" style="margin-top: 15px; color: #7f8c8d;"></p>
+        </div>
+
+        <div id="content" class="hidden">
+            <div class="section">
+                <h2>Summary Statistics</h2>
+                <div class="summary-stats" id="summaryStats"></div>
+            </div>
+
+            <div class="section">
+                <h2>1. Time Breakdown Per Iteration</h2>
+                <div style="margin-bottom: 10px;">
+                    <label><input type="checkbox" id="sortIterations" checked> Sort by most time spent</label>
+                </div>
+                <div class="chart-container">
+                    <canvas id="iterationChart"></canvas>
+                </div>
+            </div>
+
+            <div class="section">
+                <h2>2. Rule Time Breakdown by Iteration</h2>
+                <div class="iteration-selector" id="iterationSelector"></div>
+                <div style="margin-bottom: 10px;">
+                    <label><input type="checkbox" id="hideRuleNames" checked> Hide rule names on x-axis</label>
+                </div>
+                <div class="chart-container">
+                    <canvas id="ruleChart"></canvas>
+                </div>
+            </div>
+
+            <div class="section">
+                <h2>3. Query Plans</h2>
+                <div id="queryPlans"></div>
+            </div>
+        </div>
+    </div>
+
+    <script>
+        let profileData = null;
+        let iterationChart = null;
+        let ruleChart = null;
+        let selectedIteration = 0;
+        let selectedRule = null;
+        let showAllIterations = false;
+
+        document.getElementById('fileInput').addEventListener('change', handleFileUpload);
```

Comment:
Would it be too much work to hoist out a function here that takes an element, and a JSON representation of the report, and renders the editor there?

I would love to be able to embed this in the Juptyer output in Python, passing in an explicit report and render it.